Q# Katas Basic Gates Workbook. Tasks 1-4, updated

TonyHoldroyd opened the pull request:
Q# Katas Basic Gates Workbook. Tasks 1-4, updated Sunday 03/10/2021, 17:45

tcNickolas commented on BasicGates/WorkBook_BasicGates1-4_VM3.ipynb:
```diff
@@ -0,0 +1,983 @@
+    "  \\end{bmatrix}\n",
+    "  =\n",
+    "\\begin{bmatrix}\n",
+    "   1.\\alpha + 0.\\beta\\\\\n",
```
The product in the formulas is best denoted with \cdot macro

tcNickolas commented on BasicGates/WorkBook_BasicGates1-4_VM3.ipynb:
```diff
@@ -0,0 +1,983 @@
+   "id": "twenty-proportion",
+   "metadata": {},
+   "source": [
+    "This is fundamentally book work, and a well-known property of the extremely important Hadamard (H) gate.  \n",
```
This paragraph says the same thing in several different ways, but doesn't give a reminder of how the H gate looks like, like the next task. 

Generally, it would be nice to come up with a unified way to explain the solutions to the tasks in this kata that require only one gate. For example, 1) mention that a certain gate can be recognized, 2) give the matrix representation of the gate, 3) work out an example of applying this gate to some state, or something like that

(And then we get variety in the extension tasks because they each ask for things of different types)

tcNickolas commented on BasicGates/WorkBook_BasicGates1-4_VM3.ipynb:
```diff
@@ -0,0 +1,983 @@
+   "source": [
+    "And we have\n",
+    "$$\n",
+    "H(|ψ⟩) = \n",
```
Formulas written as a mix of plain text and formulas are very hard to read, and so are formulas split across multiple lines if each line ends up very short. It's better to write the whole equation as a formula, and group multiple parts of the equation in one line, so that the whole formula can be read at a glance. You can see examples in other workbooks and tutorials

tcNickolas reviewed (changes requested):
I did a pass and left some comments. They are more detailed at the beginning, because you'll be able to apply the same logic and formatting principles to the later tasks.
Also, let's call the workbook file `Workbook_BasicGates.ipynb` so that it matches our naming conventions across all workbooks.

Thank you!

TonyHoldroyd commented:
Thank you very much Maria.
So that's two sleeves and a neck line :) so far.
I'll check out your changes and make sure I follow them in my next pull request, which should be soon.
I intend compling the workbook in stages like this:
Tasks 1.5-1.7
Tasks 1.8 - 1.10
Tasks 2.1 - 2.2
Tasks 2.3 - 2.4
Those are manageable  chunks for me, hope their okay for you too, but i can always change this diivision if you like.

TonyHoldroyd commented:
Ciao Mariia,
Thank you very much.
Sorry about this, but I'm a little unsure how to proceed with my next pull
request.
I understand that what I have done so far has been merged into the master.
If I create a whole new file in my *existing* local fork containing  *just*
tasks 5-7, say, do the usual things locally , push up to my existing
github  fork and make a new pull request from there, is that what to do
please?
Thank you!
Best,
Tony



On Wed, Oct 13, 2021 at 5:23 AM Mariia Mykhailova ***@***.***>
wrote:

> ***@***.**** approved this pull request.
>
> Looks good! I made a pass cleaning up some phrasings and merging small
> cells into larger ones. For example, I removed the note about
> distinguishably different superposition in task 1, since the question of
> distinguishability of states that are not orthogonal is a tricky one, and
> definitely not the one in which we should engage the learner that early in
> their studies! I also added links from the BasicGates kata notebook to the
> workbook, so that the solutions can be already discoverable for the
> learners.
>
> Thank you!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/microsoft/QuantumKatas/pull/666#pullrequestreview-778053264>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACS4UW6CIEXXYBRQTUB5A5TUGT3S5ANCNFSM5FHXTRRQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

TonyHoldroyd commented:
Brilliant,  thank you for your rapid response Mariia

On Wed, 13 Oct 2021, 22:53 Mariia Mykhailova, ***@***.***>
wrote:

> The task grouping makes sense - it's much easier for me to review PRs that
> are only a few tasks long relatively quickly.
>
> The best thing to do is to create a branch in your fork that has the
> latest main of the microsoft repository (including the first four tasks),
> and in that branch to add new tasks to the same workbook file. Creating a
> separate file for those tasks will make for a tricky merge, and the
> workbook itself is not long enough to be split into several files logically
> (we did that for Superposition workbook, but that kata is a lot longer and
> the task explanations are huge!)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/microsoft/QuantumKatas/pull/666#issuecomment-942711561>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACS4UW6N2BJW33U3XZQBM43UGXWSXANCNFSM5FHXTRRQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>
